zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +133,137 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() { // Test multiple small transactions to fill a block max size
+    let blockheader = generate_block_header();
+
+    // 23511 transactions = data.len() = 1999925 bytes
+    // 23512 transactions = data.len() = 2000010 bytes
+    let mut limit_txs = 23511;
+
+    // Create 23511 transactions, this is the limit with our DUMMY_TX1 data
+    let mut many_transactions = Vec::new();
+    for _n in 1..limit_txs+1 {
+        let trx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]);
+        many_transactions.push(Arc::new(trx.unwrap()));
+    }
+    assert_eq!(limit_txs, many_transactions.len());
+
+    // Add the transactions into a block
+    let mut block = Block {
+        header: blockheader,
+        transactions: many_transactions.clone(),
+    };
+
+    // Serialize the block
+    let mut data = Vec::new();
+    block.zcash_serialize(&mut data).expect("block should serialize as we are not limiting generation yet");
+
+    // Check size
+    assert_eq!(data.len(), 1999925);
+
+    // Deserialize by now is ok as we are lower than the limit
+    let block2 = Block::zcash_deserialize(&data[..]).expect("block should deserialize as we are just below limit");
+
+    assert_eq!(block, block2);
+    assert_eq!(limit_txs, block2.transactions.len());
+
+    // Add 1 more transaction to the block, limit will be reached
+    limit_txs = limit_txs + 1;
+    let trx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]);
```

Comment:
assuming `Transaction` is clone you might want to create a single trx at the top of the file and then clone it everywhere you need it.

Reply:
I didn't moved it to the start of the file nor made them `const` but moved to the top of the 2 tests and cloned where needed as suggested.